### PR TITLE
bluetooth: controller: Add comments across ull

### PR DESCRIPTION
**Pull request by mped-oticon**
* Clarify certain functions and data structures in upper link layer.
 * Disambiguate node rx event done, add commentary.

`MFIFO_DEQUEUE` macro does string concatenation.
`MFIFO_DEQUEUE(done)` does not refer to `done` (the stack local), but `mfifo_done`. 
Therefore, rename to disambiguate.

No functional change expected.

Signed-off-by: Mark Ruvald Pedersen <mped@oticon.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull.c [hunk 1]
```diff
@@ -1311,7 +1357,11 @@ static void rx_demux(void *param)
 	} while (link);
 }
 
-static inline void rx_demux_rx(memq_link_t *link, struct node_rx_hdr *rx)
```
Comment (aescolar) [inline subsys/bluetooth/controller/ll_sw/ull.c]: You pressed the wrong key here :)

**Review by aescolar — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/ull.c [hunk 2]
```diff
@@ -1082,10 +1125,13 @@ static inline int init_reset(void)
 	return 0;
 }
 
-static inline void done_alloc(void)
```
Comment (aescolar) [inline subsys/bluetooth/controller/ll_sw/ull.c]: and here

File: subsys/bluetooth/controller/ll_sw/ull.c (hunk 1 above)
Comment (mped-oticon) [reply subsys/bluetooth/controller/ll_sw/ull.c]: fixed

**Review by cvinayak — APPROVED** [review]
LGTM

**Review by thoh-ot — APPROVED** [review]

**Pull request merged**